pkg/controllers/addons_controller.go

```diff
@@ -34,23 +36,40 @@ import (
 // AddonsLayerReconciler reconciles a AddonsLayer object.
 type AddonsLayerReconciler struct {
 	client.Client
-	Log     logr.Logger
-	Scheme  *runtime.Scheme
-	Applier apply.LayerApplier
+	Config   *rest.Config
+	k8client *kubernetes.Clientset
+	Log      logr.Logger
+	Scheme   *runtime.Scheme
+	Context  context.Context
+	Applier  apply.LayerApplier
 }
 
 // NewReconciler returns an AddonsLayerReconciler instance
-func NewReconciler(client client.Client, logger logr.Logger,
+func NewReconciler(config *rest.Config, client client.Client, logger logr.Logger,
```

Comment:
You could probably get away with not passing in the scheme as this is normally available on the manager `ctrl.Manager` and you could grab it in the setup func

Reply:
And i see you are getting it from the mgr when calling NewReconciler